src/Game.js

```diff
@@ -0,0 +1,213 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      player1: { name: null, id: null, paddle: [235, 480] },
+      player2: { name: null, id: null, paddle: [235, 10] },
+      pongBall: [250, 250],
+      ballDirection: [-1, -1],
+      round: 0,
+      gameCode: null,
+    };
+  }
+
+  updateAnimationState = () => {
+    this.animateScene();
+    requestAnimationFrame(this.updateAnimationState);
+  };
+
+  createPlayer = (username, id, x, y) => {
+    const player = {
+      username: username,
+      id: id,
+      score: 0,
+      paddle: [x, y],
+    };
+    if (this.state.player1) {
+      this.setState({
+        player2: player,
+      });
+    } else {
+      this.setState({
+        player1: player,
+      });
+    }
+  };
+
+  createGameSettings = (x, y) => {
+    const pongBall = [x, y];
+    this.setState({ pongBall: pongBall });
+    // setGameCode(gameCode);
+  };
+
+  createGame = () => {
+    this.createGameSettings(250, 100);
+    this.createPlayer("Jon", 1, 235, 480);
+    this.createPlayer("Joe", 2, 235, 10);
+  };
+
+  startGame = () => {
+    this.createGame();
+    requestAnimationFrame(this.updateAnimationState);
+  };
+
+  animateScene = () => {
+    this.detectCollision();
+    this.incrementBallPosition();
+  };
+
+  incrementBallPosition = () => {
+    this.setState((prevState) => ({
+      pongBall: [
+        prevState.pongBall[0] + prevState.ballDirection[0],
+        prevState.pongBall[1] + prevState.ballDirection[1],
+      ],
+    }));
+  };
+
+  handleKeydown = (event) => {
+    event.preventDefault();
+    const key = event.key;
+    if (key === "ArrowLeft") {
+      this.setState((prevState) => ({
+        player1: {
+          ...prevState.player1,
+          paddle: [
+            prevState.player1.paddle[0] - 15,
+            prevState.player1.paddle[1],
+          ],
+        },
+      }));
+    } else if (key === "ArrowRight") {
+      this.setState((prevState) => ({
+        player1: {
+          ...prevState.player1,
+          paddle: [
+            prevState.player1.paddle[0] + 15,
+            prevState.player1.paddle[1],
+          ],
+        },
+      }));
+    } else if (key === "z") {
+      this.setState((prevState) => ({
+        player2: {
+          ...prevState.player2,
+          paddle: [
+            prevState.player2.paddle[0] - 15,
+            prevState.player2.paddle[1],
+          ],
+        },
+      }));
+    } else if (key === "x") {
+      this.setState((prevState) => ({
+        player2: {
+          ...prevState.player2,
+          paddle: [
+            prevState.player2.paddle[0] + 15,
+            prevState.player2.paddle[1],
+          ],
+        },
+      }));
+    }
+  };
+
+  detectCollision = () => {
+    let [player1PaddleX, player1PaddleY] = this.state.player1.paddle;
+    let [player2PaddleX, player2PaddleY] = this.state.player2.paddle;
+    let [ballX, ballY] = this.state.pongBall;
+
+    let player1RightEdge = player1PaddleX + 30;
```

Comment:
assuming that 30 is the pixel length of the paddle? 

We should capture that in a static property that needs only be updated in one spot vs. tracking down anywhere we are hard coding the paddle width/height